src/document/DocumentAnnotator.ts

```diff
@@ -99,27 +123,25 @@ export default class DocumentAnnotator extends BaseAnnotator {
     };
 
     handleSelectionChange = (): void => {
-        // Clear previous selection immediately
-        this.store.dispatch(setSelectionAction(null));
-        clearTimeout(this.selectionChangeTimer);
+        if (this.isMouseSelecting) {
+            return;
+        }
 
-        this.selectionChangeTimer = window.setTimeout(() => {
-            this.store.dispatch(setSelectionAction(getSelection()));
-        }, TEXT_LAYER_ENHANCEMENT);
+        this.store.dispatch(setSelectionAction(getSelection()));
     };
 
-    debounceHandleSelectionChange = debounce(this.handleSelectionChange, SELECTION_CHANGE_DEBOUNCE, {
-        // The previous selection needs to be cleared immediately when selection changes
-        // Below options make sure the function is triggered on the leading edge of the timeout,
-        // instead of on the trailing edge
-        leading: true,
-        trailing: false,
-    });
+    debounceHandleSelectionChange = debounce(this.handleSelectionChange, SELECTION_CHANGE_DEBOUNCE);
 
     render(): void {
         // Clear previous selection
         this.store.dispatch(setSelectionAction(null));
 
+        if (!this.mouseEventHandlerAdded && this.annotatedEl) {
```

Comment:
Would an overridden `init` method in `DocumentAnnotator` be a better place to add this?

Reply:
I think we should try to avoid overriding `init`. It calls `render`, so it doesn't seem to change much in this circumstance.